src/safeposix/syscalls/net_calls.rs

```diff
@@ -337,27 +406,40 @@ impl Cage {
         0
     }
 
+    //Helper function of bind_syscall
+    //Task: checks if fd refers to a valid socket file descriptor
+    //fd: the file descriptor associated with the socket
+    //localaddr: reference to the GenSockaddr enum that hold the address
+    //prereserved: bool that describes whether the address and port have 
+    //             been set aside or designated for specific purposes
     pub fn bind_inner(
         &self,
         fd: i32,
         localaddr: &interface::GenSockaddr,
         prereserved: bool,
     ) -> i32 {
+        //chekedfd is an atomic reference count of a lock on the fd
         let checkedfd = self.get_filedescriptor(fd).unwrap();
-        let mut unlocked_fd = checkedfd.write();
+        //returns a write lock once no other writers or readers have access to the lock
+        let mut unlocked_fd = checkedfd.write(); //why isn't .unwrap() used here?
         if let Some(filedesc_enum) = &mut *unlocked_fd {
             match filedesc_enum {
+                //*unlocked_fd is in the format of Option<T>, where T is of type Socket(&mut SocketDesc)
                 Socket(ref mut sockfdobj) => {
+                    //Clone the socket handle
                     let sock_tmp = sockfdobj.handle.clone();
+                    //Access write gaurd for socket handle
```

Comment:
```suggestion
                    //Access write guard for socket handle
```

Does this access the write guard?  How?